crates/ruff/src/rules/flake8_pyi/rules/prefix_type_params.rs

```diff
@@ -83,8 +90,19 @@ pub(crate) fn prefix_type_params(checker: &mut Checker, value: &Expr, targets: &
         }) else {
             return;
         };
-        checker
-            .diagnostics
-            .push(Diagnostic::new(UnprefixedTypeParam { kind }, value.range()));
+        let mut diagnostic = Diagnostic::new(UnprefixedTypeParam { kind }, value.range());
+        if checker.patch(Rule::UnprefixedTypeParam) {
+            if let Expr::Name(ast::ExprName { range, .. }) = &targets[0] {
+                diagnostic.set_fix(Fix::automatic_edits(
+                    Edit::insertion(format!("_"), range.start()),
+                    [Edit::insertion(
+                        format!("_"),
+                        args[0].range().start() + TextSize::from(1),
+                    )],
+                ));
+            };
+        };
```

Comment:
If the type parameter is referenced elsewhere in the file, this change would break those references, right?

Reply:
Hmmm, you are correct. Fixing that might be non-trivial. Will think on it. Setting this PR to draft for now and will probably separate the two autofixes, and implement Y048 in between.

Reply:
We actually do now have the ability to track all read-references of a given symbol, so we could support this, but it would require some further changes (e.g., we'd need to run this rule _after_ visiting the entire file). I might use it as an excuse to figure out how to make it work.

Reply:
Separated, making this PR only for PYI010.  Not enough brainspace this evening to come up with a solution, but no need to block the other autofix. If the overhead of more but smaller PRs is fine for you and fellow maintainers, I'll cut my contributions into single chunks to avoid too much PR scope change in the future.